gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java

```diff
@@ -683,7 +683,8 @@ public static Builder builder() {
   BucketInfo update(BucketInfo bucketInfo, BucketTargetOption... options);
 
   /**
-   * Update blob information.
+   * Update blob information. All the blob's metadata will be replaced using values from the given
+   * {@code blobInfo}. Update will fail if an incomplete {@code blobInfo} is provided.
```

Comment:
I was surprised as well to see this failing.

For `update` also `acl[]` [is required](https://cloud.google.com/storage/docs/json_api/v1/objects/update), which makes providing a default value a bit trickier.